src/load_inputs/load_generators_data.jl

```diff
@@ -185,44 +185,52 @@ function load_generators_data!(setup::Dict, path::AbstractString, inputs_gen::Di
-		inputs_gen["C_Fuel_per_MWh"][g,:] = fuel_costs[fuel_type[g]].*heat_rate[g]
-		gen_in[g,:CO2_per_MWh] = fuel_CO2[fuel_type[g]]*heat_rate[g]
-		gen_in[g,:CO2_per_MWh] *= scale_factor
-		# kton/MMBTU * MMBTU/MWh = kton/MWh, to get kton/GWh, we need to mutiply 1000
 		if g in inputs_gen["COMMIT"]
-			# Start-up cost is sum of fixed cost per start plus cost of fuel consumed on startup.
-			# CO2 from fuel consumption during startup also calculated
-
-			inputs_gen["C_Start"][g,:] = gen_in[g,:Cap_Size] * (fuel_costs[fuel_type[g]] .* start_fuel[g] .+ start_cost[g])
-			# No need to re-scale C_Start since Cap_size, fuel_costs and start_cost are scaled When Setup[ParameterScale] =1 - Dharik
-			gen_in[g,:CO2_per_Start]  = gen_in[g,:Cap_Size]*(fuel_CO2[fuel_type[g]]*start_fuel[g])
-			gen_in[g,:CO2_per_Start] *= scale_factor
-			# Setup[ParameterScale] =1, gen_in[g,:Cap_Size] is GW, fuel_CO2[fuel_type[g]] is ktons/MMBTU, start_fuel is MMBTU/MW,
-			#   thus the overall is MTons/GW, and thus gen_in[g,:CO2_per_Start] is Mton, to get kton, change we need to multiply 1000
-			# Setup[ParameterScale] =0, gen_in[g,:Cap_Size] is MW, fuel_CO2[fuel_type[g]] is tons/MMBTU, start_fuel is MMBTU/MW,
-			#   thus the overall is MTons/GW, and thus gen_in[g,:CO2_per_Start] is ton
+			# Start-up cost is sum of fixed cost per start startup.
+			inputs_gen["C_Start"][g,:] .= gen_in[g,:Cap_Size] * ( start_cost[g])
 		end
 	end
 
 	load_vre_stor_data!(inputs_gen, setup, path)
+
+	
+    # write zeros if col names are not in the gen_in dataframe
+	missing_cols = ["Biomass", "CO2_Capture_Rate", "CO2_Capture_Rate_Startup", "CO2_Capture_Cost_per_Metric_Ton"]
+	write_zeros_if_not_exist!(gen_in, missing_cols)
+
+    # Scale CO2_Capture_Cost_per_Metric_Ton for CCS units 
+	gen_in.CO2_Capture_Cost_per_Metric_Ton /= scale_factor
+
+    # Piecewise fuel usage module
+	# Users should specify how many segements are used to build piecewise fuel comsumption.
+	# Users should at least provide Slope1 and Intercept1 if they want to use piecewise fuel usage
+	if setup["UCommit"] > 0
+        if setup["PiecewiseFuelUsage"] == 1
+			num_segments = maximum(gen_in[!,:NUM_SEGMENTS])
+			inputs_gen["MAX_NUM_SEGMENTS"] = num_segments
+			# thermal generators (commit) that have NUM_SEGMENTS >= 1 are able to use piecewise fuel usage optional
+			inputs_gen["THERM_COMMIT_PWFU"] = intersect(gen_in[gen_in.THERM.==1,:R_ID], gen_in[gen_in.NUM_SEGMENTS .> 0,:R_ID])
+			# create col names based on maximum num of segments
+			slope_cols =  [ Symbol(string("Slope", i)) for i in 1:num_segments]
+			intercept_cols =  [ Symbol(string("Intercept", i)) for i in 1:num_segments]
+            # no need to scale slope, but the intercept of fuel usage in each segment needs to be scaled (MMBTU -> Billion BTU).
+			for i in 1:num_segments
+				gen_in[!, intercept_cols[i]] /= scale_factor
+			end
+			inputs_gen["slope_cols"] = slope_cols
+			inputs_gen["intercept_cols"] = intercept_cols
+		end
+	end
```

Comment:
Can this whole portion be moved to a new function which is called in `load_generators_data`?

Reply:
maybe `process_piecewisefuelusage`

Reply:
done!